plugins/uefi-capsule/fu-uefi-cod-device.c

```diff
@@ -134,6 +134,39 @@ fu_uefi_cod_device_get_results(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gchar *
+fu_uefi_cod_device_get_indexed_filename(FuUefiDevice *self, GError **error)
+{
+	g_autofree gchar *esp_path = fu_volume_get_mount_point(fu_uefi_device_get_esp(self));
+	for (guint i = 0; i < 0xFFFF; i++) {
+		g_autofree gchar *basename = g_strdup_printf("CapsuleUpdateFile%04X.bin", i);
+		g_autofree gchar *cod_path =
+		    g_build_filename(esp_path, "EFI", "UpdateCapsule", basename, NULL);
+		if (!g_file_test(cod_path, G_FILE_TEST_EXISTS))
+			return g_steal_pointer(&cod_path);
+	}
+	g_set_error_literal(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_INTERNAL,
+			    "cannot find capsule update file idx that does not exist");
```

Comment:
If this ever happens; this is a really confusing message.
```suggestion
			    "all potential CapsuleUpdateFile file names are taken");
```